draft-ietf-quic-transport.md

```diff
@@ -725,28 +725,57 @@ QUIC encodes the transport parameters and options as tag-value pairs, all as
-QUIC uses a cookie, called the Source Address Token (STK), to mostly eliminate
-this roundtrip of delay.  This technique is similar to TCP Fast Open's use of a
-cookie to avoid a roundtrip of delay in TCP connection establishment.
-
-On a new connection, a QUIC server sends an STK, which is opaque to and stored
-by the client.  On a subsequent connection, the client echoes it in the
-transport handshake as proof of IP ownership.
-
-A QUIC server also uses the STK to store server-designated connection IDs for
-Stateless Rejects, to verify that an incoming connection contains the correct
-connection ID.
-
-A QUIC server MAY additionally store other data in a the STK, such as measured
-bandwidth and measured minimum RTT to the client that may help the server better
-bootstrap a subsequent connection from the same client.  A server MAY send an
-updated STK message mid-connection to update server state that is stored at the
-client in the STK.
-
-(TODO: Describe server and client actions on STK, encoding, recommendations for
-what to put in an STK.  Describe SCUP messages.)
+Transport protocols commonly spend a round trip checking that a client owns the
+transport address (IP and port) that it claims.  Verifying that a client can
+receive packets sent to its claimed transport address protects against spoofing
+of this information by malicious clients.
+
+This technique is used primarily to avoid QUIC from being used for traffic
+amplification attack.  In such an attack, a packet is sent to a server with
+spoofed source address information that identifies a victim.  If a server
+generates more or larger packets in response to that packet, the attacker can
+use the server to send more data toward the victim than it would be able to send
+on its own.
+
+Several methods are used in QUIC to mitigate this attack.  Firstly, the initial
+handshake packet from a client is padded to a moderately large size (TBD:
+describe/reference how this size is selected).  This allows a server to send a
+similar amount of data without validating ownership of an address (TBD: provide
+limits on what amount of amplification is enough).
+
+A server eventually confirms that a client has received its messages when the
+cryptographic handshake successfully completes.  This might be either because
+the server wishes to avoid the computational cost of completing the handshake,
+or it might be that the size of the packets that are sent during the handshake
+is too large.  This is especially important for 0-RTT, where the server might
+wish to provide application data traffic - such as a response to a request - in
+response to the data carried in the early data from the client.
+
+To send additional data prior to completing the cryptographic handshake, the
+server then needs to validate that the client owns the address that it claims.
+
+Two tools are provided by TLS to enable validation of client source addresses:
```

Comment:
The ticket would only be used for 0-RTT.  You don't get to use the cookie extension for the first ClientHello.  That's a TLS limitation, we don't have a mechanism for providing a cookie for use in a future connection.  Nor do we need one because that is something that the session ticket can be used for.